### ISSUE TITLE
How should the absorption of photons and subsequent re-emission be visually portrayed?

### ISSUE THREAD

**Issue by jbphet** [phetsims/greenhouse-effect#167]
As of this writing, the code can support making photons disappear when absorbed in a layer and make a lateral jump when re-emitted.  The length of time that the photon is absorbed and the max distance it can jump are configurable.  However, pretty much every variation we've tried so far turns out to be visually hard to follow, see https://github.com/phetsims/greenhouse-effect/issues/163.  In today's design meeting we discussed some other possibilities, such as:

- fade out/in
- shrink/grow
- leave a bit of a glowing spot

So far, we're not enamored of any of these, so we decided to log an issue and sort through some thoughts and possibly prototype them once they are a bit more refined.

Assigning to @arouinfar for now for brainstorming.

**Comment by arouinfar** [comment]
I think it's time to revisit this in a design meeting. With the current implementation (see latest [dev version](https://phet-dev.colorado.edu/html/greenhouse-effect/1.1.0-dev.14/phet/greenhouse-effect_en_phet.html)) the photons are generally easy to track. However, it's hard to detect when a photon is absorbed and emitted upward since there's only a slight effect on the direction of travel. I think we could try two things:
* Adjust photon opacity to fade in/out to indicate a photon has been absorbed.
* Impose a minimum deflection angle of upwardly emitted photons so it's not too near the normal.

**Comment by pixelzoom** [comment]
This stalled out ~8 months ago.  @arouinfar Is this still an issue? Does this still need a design meeting?

**Comment by arouinfar** [comment]
Thanks for the ping @pixelzoom. My previous comment still holds -- upwardly emitted photons can be hard to track. Sometimes the deflection angle is so small it appears to travel through the layer unperturbed. I think a good first step would be to try imposing a minimum deflection angle to see how things look. @jbphet can you give it a go?

**Comment by jbphet** [comment]
I've implemented a minimum deflection amount and chose a value based on my own eye.  This can be easily tweaked if needed, though it can't go a _lot_ higher without also tweaking the "direction cone" in which photons are allowed to travel.

@arouinfar - please have a look and see what you think.  If more refinement is needed we should set up a design meeting to tweak it to our satisfaction in real time.

**Comment by pixelzoom** [comment]
Looks like this should have the "ready-for-review" label, so I'll add it.

**Comment by jbphet** [comment]
@kathy-phet, @arouinfar, and I just met and reviewed this, and decided to add back a bit of absorption time for the photons on the "Photons" and "Layer Model" screens.  After experimenting with it for a bit, we settled on 0.1 seconds of photon absorption time in each layer for the "Photons" screen and 0.25 seconds for the "Layer Model" screen.  We will keep the lateral jump at zero, since this makes it hard to track the photons.

**Comment by jbphet** [comment]
Signed off on by the design team, closing.

**Comment by phet-dev** [comment]
Reopening because there is a TODO marked for this issue.

**Comment by jbphet** [comment]
I've addressed to TODO that caused this to be re-opened by making the layers essentially have zero thickness for their photon absorption behavior, since that was essentially what was approved by the design team.  Closing.
